packages/gatsby/src/cache-dir/production-app.js

```diff
@@ -77,11 +77,13 @@ window.___navigateTo = navigateTo
 const history = createHistory()
 
 function attachToHistory(history) {
-  window.___history = history
+  if(!window.___history) {
+    window.___history = history
 
-  history.listen((location, action) => {
-    apiRunner(`onRouteUpdate`, { location, action })
-  })
+    history.listen((location, action) => {
+      apiRunner(`onRouteUpdate`, { location, action })
+    })
+  }
```

Comment:
Updated, also noticed that this same code exists within: 
https://github.com/gatsbyjs/gatsby/blob/08bddd92b60bccd125b2c147da4ec896f48197c3/packages/gatsby/src/cache-dir/root.js#L104

But the production-app.js file is the root cause